### PR TITLE
Fix a bug in ImportScannerImpl.visitCtInvocation

### PR DESCRIPTION
**Pull request by leventov**
shouldn't import invoked method declaration classes

### CODE REVIEW

File: src/main/java/spoon/reflect/visitor/ImportScannerImpl.java [hunk 1]
```diff
@@ -63,12 +63,10 @@
 
 	@Override
 	public <T> void visitCtInvocation(CtInvocation<T> invocation) {
-		// For a ctinvocation, we don't have to import declaring type
 		enter(invocation);
 		scan(invocation.getAnnotations());
 		scanReferences(invocation.getTypeCasts());
 		scan(invocation.getTarget());
-		scan(invocation.getExecutable());
```
Comment (tdurieux) [inline src/main/java/spoon/reflect/visitor/ImportScannerImpl.java]: I already try to remove this line but some time this line is required.
I don't remember in which case (may be in static call).

**Comment by GerardPaligot** [comment]
@leventov Can you add a test case where it is useful to avoid scanning executable of an invocation?

**Comment by leventov** [comment]
In my case this tries to import package-private class, if the called method is originally defined in it, e. g.

```java
package foo;
class Foo {
    public void foo() {}
}

public class Bar extends Foo {
}
```

```java
package other.package;

bar.foo();
```

When spoon compiles imports for such, it includes `import foo.Foo;` that is illegal.

File: src/main/java/spoon/reflect/visitor/ImportScannerImpl.java (hunk 1 above)
Comment (leventov) [inline src/main/java/spoon/reflect/visitor/ImportScannerImpl.java]: @tdurieux static call target (class name) should be processed by the prev line `scan(invocation.getTarget())`, isn't it?

Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/ImportScannerImpl.java]: yes, `getTarget()` is used for static invocation.

**Comment by GerardPaligot** [comment]
@leventov I made a PR on your repository. Can you accept it to merge your PR here?

**Comment by leventov** [comment]
@GerardPaligot thank for making that ground work instead of me :)

**Comment by GerardPaligot** [comment]
No problem @leventov :)

@monperrus I contributed on this PR. I let you merge.

**Pull request merged**